vajram/vajram-samples/src/test/java/com/flipkart/krystal/vajram/samples/benchmarks/calculator/FormulaTest.java

```diff
@@ -227,12 +237,105 @@ private static CompletableFuture<Integer> asyncFormula(int value) {
+        vajramID(getVajramIdString(Adder.class)),
+        InputModulatorConfig.simple(() -> new Batcher<>(100)));
+    KryonExecutorConfigBuilder kryonExecutorConfigBuilder = KryonExecutorConfig.builder()
+        .kryonExecStrategy(
+            KryonExecStrategy.BATCH).graphTraversalStrategy(GraphTraversalStrategy.DEPTH);
+    try (KrystexVajramExecutor<FormulaRequestContext> krystexVajramExecutor =
+        graph.createExecutor(new FormulaRequestContext(100, 20, 5, REQUEST_ID),
+            VajramTestHarness
+                .prepareForTest(kryonExecutorConfigBuilder)
+                .withMock(AdderRequest.builder().numberOne(20).numberTwo(5).build(), ValueOrError.withValue(25))
+                .buildConfig())) {
+      future = executeVajram(krystexVajramExecutor, 0);
+    }
+    assertThat(future.get()).isEqualTo(4);
+    assertThat(Adder.CALL_COUNTER.sum()).isEqualTo(0);
+  }
+
+  @Test
+  void formula_success_with_mockedDependencyDivider() throws Exception {
+    CompletableFuture<Integer> future;
+    VajramKryonGraph graph = this.graph.build();
+    graph.registerInputModulators(
+        vajramID(getVajramIdString(Adder.class)),
+        InputModulatorConfig.simple(() -> new Batcher<>(100)));
+    KryonExecutorConfigBuilder kryonExecutorConfigBuilder = KryonExecutorConfig.builder()
+        .kryonExecStrategy(
+            KryonExecStrategy.BATCH).graphTraversalStrategy(GraphTraversalStrategy.DEPTH);
+    try (KrystexVajramExecutor<FormulaRequestContext> krystexVajramExecutor =
+        graph.createExecutor(new FormulaRequestContext(100, 20, 5, REQUEST_ID),
+            VajramTestHarness
+                .prepareForTest(kryonExecutorConfigBuilder)
+                .withMock(DividerRequest.builder().numerator(100).denominator(25).build(), ValueOrError.withValue(4))
+                .buildConfig())) {
+      future = executeVajram(krystexVajramExecutor, 0);
+    }
+    assertThat(future.get()).isEqualTo(4);
+    assertThat(Adder.CALL_COUNTER.sum()).isEqualTo(1);
+  }
+
+  @Test
+  void formula_failure() throws Exception {
+    CompletableFuture<Integer> future;
+    VajramKryonGraph graph = this.graph.build();
+    graph.registerInputModulators(
+        vajramID(getVajramIdString(Adder.class)),
+        InputModulatorConfig.simple(() -> new Batcher<>(100)));
+    KryonExecutorConfigBuilder kryonExecutorConfigBuilder = KryonExecutorConfig.builder()
+        .kryonExecStrategy(
+            KryonExecStrategy.BATCH).graphTraversalStrategy(GraphTraversalStrategy.DEPTH);
+    try (KrystexVajramExecutor<FormulaRequestContext> krystexVajramExecutor =
+        graph.createExecutor(new FormulaRequestContext(100, 0, 0, REQUEST_ID),
+            VajramTestHarness
+                .prepareForTest(kryonExecutorConfigBuilder)
+                .withMock(AdderRequest.builder().numberOne(0).numberTwo(0).build(),
+                    ValueOrError.withValue(0))
+                .buildConfig())) {
+      future = executeVajram(krystexVajramExecutor, 0);
```

Comment:
We should assert on the result of this future to make sure that it has failed as expected